### ISSUE TITLE
Provide a way for offline cross-signing verification

### ISSUE THREAD

**Issue by vit9696** [vector-im/riot-web#13931]
**Is your suggestion related to a problem? Please describe.**
Currently it is not possible to perform offline participation in cross-signing, which you can only do simultaneously with e.g. emojis. This does not work in a number of use-cases for the same reasons I explained in https://github.com/vector-im/riot-web/issues/12589.

Furthermore, if you verify sessions one by one you basically end up in a situation when you cannot differentiate two cases as they both show **black shield** icons:

- you verified all the session keys (and the chat is somewhat trusted) 
- you verified some (or no) session keys (and the chat is not trusted)

This to my eyes is in fact a security risk and usability breakage upon migrating from 1.5.x to 1.6.x.

**Describe the solution you'd like**
There are two parts of the solution here:

1. Differentiate `all sessions verified` and `not all sessions verified` cases by providing a different shield icon. Either green or yellow for `all sessions verified`. This is particularly reasonable for people who do not want to trust cross-signing.

2. Make it possible to verify users on offline basis by manually comparing the cross-signing public key fingerprints all the same way one could manually verify the sessions (just like with https://github.com/vector-im/riot-web/issues/12589)

    I.e. I can get a key for the person I want to talk to, compare it myself at the time I want, accept it in Riot to allow that person to later add more devices. It should also be possible for me to transfer that key to the other person, so that I do not have to ask each person to send me the keys.

**Comment by vit9696** [comment]
CC'ing @jryans due to the previous issue participation. Also linking to https://github.com/vector-im/riot-web/issues/12806, which is related but different from this one as it talks about users without cross-signing not users with cross-signing but verified/unverified sessions.

**Comment by jryans** [comment]
Thanks for the feedback, we'll track this as we continue evaluating and evolving the space of supported use cases.

Related also to https://github.com/vector-im/riot-web/issues/12719.

**Comment by maisiliymBridj** [comment]
Users verify each other's cross-signing public key, right? So where is that public key? The questionner wants to put it on his contact card. If the questionner has another's public key, how does he enter it into his client? This is public-keys verification 101. Do matrix devs know about public-key verification 101? If not, what is their excuse for missing such an obvious point?

**Comment by jryans** [comment]
Yes, there's (several) public / private key pairs. They are currently not exposed in UI because the initial expectation was that most users would not verify they directly by comparing keys, but instead via an online method like the emoji comparison or QR code scanning. In such a method, both sides trade verification messages that include key data as well as performing an OOB comparison.

**Comment by ara4n** [comment]
The fact we've lost the ability to do manual public fingerprint verification is very unfortunate when working with clients which haven't yet implemented interactive verification.

**Comment by t3chguy** [comment]
We haven't lost that though?
You can select a user's device and verify it via Text

![image](https://user-images.githubusercontent.com/2403652/99890556-71079a80-2c58-11eb-8597-70119d331ff5.png)

**Comment by ara4n** [comment]
hm, i must be going blind. in which case, i don't understand this bug...

**Comment by vit9696** [comment]
This thing is per-device. It does not let you validate _cross-signing_. And in fact, I am pretty even per device textual verification had had numerous issues with different clients for quite some time, and part two of the issue should still be valid as well.
